### DISCUSSION TITLE
next/router gives error

### DISCUSSION THREAD

**Discussion by callumok2004** [vercel/next.js#26516] [Help]
This video will pretty much break down the issue, sometimes next will push the new route, without an issue, other times the route is not pushed and an error is shown.

![jty7o 1](https://user-images.githubusercontent.com/35779365/123058048-5240d080-d400-11eb-968d-9fe829258d7c.gif)

This is the code I am using
```ts
function redirect(e, num: string) {
        e.preventDefault();
        setNav(num);
        setPage(e.target.href);
        router.push(e.target.href);
    }
```
the `a` tag has `onClick={(e) => redirect(e, "one")}` and `href="/"` / `href="/friends", I have also tried passing the second parameter of the push with the href, and adding `{shallow: true}` but nothing works. It is strange how the error only occurs sometimes, and seems to be completely random, though more frequent than a successful push.

**Comment by adebiyial** [comment]
First, what are you trying to achieve? Move between different pages?

Why not use the `Link` component?

**Reply by callumok2004** [reply]
I do not see any errors in the console, as you asked before editing. And I am not using a Link component as I am having two global context items being updated before the route is pushed.

**Reply by callumok2004** [reply]
Also, if I try to use `Link`, I'm receiving yet another error.
![image](https://user-images.githubusercontent.com/35779365/123164154-7afdb000-d46a-11eb-8dd6-c5e264a31a16.png)

**Reply by callumok2004** [reply]
Seems like globalContext doesn't work properly when using `Link`

**Reply by adebiyial** [reply]
Ah. You caught my message before the edit.

Also, I think you should let `Link` do its job and find a way to update your global context.

The context should be updated with the new route? Why not try to update in `_app.js`

**Comment by callumok2004** [comment]
Update: noticed sometimes element.target is not even the `a` element???
![image](https://user-images.githubusercontent.com/35779365/123165679-70dcb100-d46c-11eb-9b9e-7d57599e4e43.png)
![image](https://user-images.githubusercontent.com/35779365/123165726-818d2700-d46c-11eb-9a93-a83bef09b23a.png)
![image](https://user-images.githubusercontent.com/35779365/123165767-8c47bc00-d46c-11eb-96aa-4e80b01a1c39.png)
sometimes its the svg??
